docs/community/maintainers/release-process.md

```diff
@@ -37,27 +37,72 @@ YYYY-MM-releaseNumber
+
+### Release Process
+
+The Nebari release process is a structured workflow that ensures the quality and reliability of each release. The process consists of the following key steps:
+
+1. **Preparation**: Identify the changes to include in the release and associate them
+   with the appropriate milestone. Assign a Nebari core maintainer as the "Release
+   Captain" and open a release checklist issue to track the process.
+
+2. **Release Candidate**: After merging all necessary pull requests, the Release Captain
+   creates an initial Release Candidate using the pre-release option in the GitHub
+   releases panel. A review checklist issue is also opened to ensure that all changes
+   are correctly included in the release. This could be a iterative process, with
+   multiple release candidates being created in case of new changes or fixes being needed.
+
+3. **Testing and Verification**: Complete the review checklist to confirm that all
+   changes function as expected and that major services perform optimally. The Release
+   Captain assigns owners to each checklist item to ensure accountability.
 
-- `YYYY-MM-releaseNumber` - Represents the tag for a particular release.
+4. **Finalizing the Release**: Upon successful testing, the Release Captain updates the
+   release notes and increments the version number in the `constants.py` file. The final
+   release is then published, and the release checklist issue is updated with the final
+   details.
 
-### Process
+For a detailed guide on the release process, refer to the [release checklist template](https://github.com/nebari-dev/nebari/blob/main/.github/ISSUE_TEMPLATE/release-checklist.md).
 
-The release process is captured in the [release checklist template](https://github.com/nebari-dev/nebari/blob/main/.github/ISSUE_TEMPLATE/release-checklist.md). In the event that a patch or hotfix release is needed, release process is the same as outlined above.
+### Hotfixes and Patch Releases
 
-## Related packages
+If a patch or hotfix release is necessary, the process mirrors the standard release steps with a key difference:
+
+- The Release Captain creates a new **hotfix branch** (see the convention above) from the previous release **tag**.
+- Necessary changes are cherry-picked into this hotfix branch.
+- The final release is published following the standard procedure.
+
+This approach ensures that critical fixes are efficiently addressed without disrupting
+the main development workflow.
+
+:::note
+Hotfix releases are rare and are only used to address critical issues that cannot wait
+for the next scheduled release, they are usually associated with
+[broken](https://conda-forge.org/docs/maintainer/updating_pkgs/#removing-broken-packages)
+or [yanked](https://pypi.org/help/#yanked) releases. And should be delivered as soon as possible.
+:::
```

Comment:
Need to include dedicated section about broken releases, and how to proceed in each mirror